File: simple_LR.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Mon Nov  6 20:12:02 2017

@author: shiwani 
"""

# Data Preprocessing Template

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

# Splitting the dataset into the Training set and Test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)

# Feature Scaling will be taken care of by the libaray for simple linear regression

# fitting simple linear regression to the training set, created a machine regressor of class LinearRegression and made it learn the traiining set with the help of a method 

from sklearn.linear_model import LinearRegression 
regressor = LinearRegression()
regressor.fit(X_train,y_train)

#predicting the test set results

y_pred=regressor.predict(X_test)

#visualising

plt.scatter(X_train,y_train,color='red')
plt.plot(X_train,regressor.predict(X_train),color='blue')
plt.xlabel('years of experience')
plt.ylabel('salary')
plt.show

plt.scatter(X_test,y_test,color='red')
plt.plot(X_train,regressor.predict(X_train),color='blue')
plt.xlabel('years of experience')
plt.ylabel('salary')
plt.show

File: Simple_LR.r
#Simple linear regression

dataset=read.csv('Salary_Data.csv')



#splitting the data set into training and test sets
#install.packages('caTools')
library(caTools)
set.seed(123)
split=sample.split(dataset$Salary, SplitRatio = 2/3)
training_set=subset(dataset,split==TRUE)
test_set=subset(dataset,split==FALSE)

#fitting SLR to the training set

regressor= lm(formula=Salary ~ YearsExperience, data = training_set)

#predicting test results
 y_pred=predict(regressor,newdata=test_set)
 
 #visualising install.packages('ggplot2')

library(ggplot2)
 
ggplot()+
  geom_point(aes(x= training_set$YearsExperience, training_set$Salary), 
             colour= 'red')+
  geom_line(aes(x= training_set$YearsExperience,y= predict(regressor,newdata=training_set)),
            colour='blue') +
  ggtitle('salary vs exp') +
  xlab('years of experience') +
  ylab('salary')

ggplot()+
  geom_point(aes(x= test_set$YearsExperience, test_set$Salary), 
             colour= 'red')+
  geom_line(aes(x= training_set$YearsExperience,y= predict(regressor,newdata=training_set)),
            colour='blue') +
  ggtitle('salary vs exp') +
  xlab('years of experience') +
  ylab('salary') 